src/gov/nist/javax/sip/SipProviderImpl.java

```diff
@@ -491,8 +492,25 @@ public ServerTransaction getNewServerTransaction(Request request)
         SIPRequest sipRequest = (SIPRequest) request;
         try {
             sipRequest.checkHeaders();
-        } catch (ParseException ex) {
-            throw new TransactionUnavailableException(ex.getMessage(), ex);
+        } catch (ParseException pe) {
+            try {
+                SIPResponse badReqRes = sipRequest.createResponse(Response.BAD_REQUEST,
+                        "Bad Request (" + pe.getLocalizedMessage() + ')');
+                // Should add a to-tag if not already present...
+                ToHeader to = (ToHeader) badReqRes.getHeader(ToHeader.NAME);
+                if (to.getTag() == null || to.getTag().isEmpty()) {
+                    to.setTag("badreq");
+                    badReqRes.setHeader(to);
+                }
+                if (logger.isLoggingEnabled(LogWriter.TRACE_DEBUG)) {
+                    logger.logDebug("Sending automatic 400 Bad Request:");
+                    logger.logDebug(badReqRes.toString());
+                }
+                sendResponse(badReqRes);
```

Comment:
Implementation is based on MessageChannel.createBadReqRes, so we plan to add it, it should be added everywhere
We can remove exception message from the response:
  SIPResponse badReqRes = sipRequest.createResponse(Response.BAD_REQUEST, "Bad Request (" + pe.getLocalizedMessage() + ')');
And add it into the Reason-Phrase header